src/Unicorn/SerializationHelper.cs

```diff
@@ -32,6 +32,8 @@ namespace Unicorn
 	/// </summary>
 	public class SerializationHelper
 	{
+		public Dictionary<string, object> PipelineArgumentData { get; set; }
```

Comment:
For improved developer experience I'd suggest making this default to `new Dictionary<string, object>()` and removing the public setter - then it's no longer needed to check for null values in merge, and it's not needed to initialize the dictionary prior to using it.